src/connector/src/lib.rs

```diff
@@ -136,4 +136,375 @@ mod tests {
 
         expect_file!("../with_options_sink.yaml").assert_eq(&generate_with_options_yaml_sink());
     }
+
+    /// Test some serde behavior we rely on.
```

Comment:
Actually we didn't use `deny_unknown_fields` now. Not sure whether to keep the tests. Some related with `flatten` might still be useful.